subsys/bluetooth/controller/ll_sw/nordic/lll/lll_central_iso.c

```diff
@@ -1237,3 +1133,147 @@ static void isr_done(void *param)
+
+			/* sn and nesn are 1-bit, only Least Significant bit is needed */
+			cis_lll->nesn += nesn_update;
+			cis_lll->rx.payload_count += cis_lll->rx.bn;
+			cis_lll->rx.bn_curr = 1U;
+		}
+	}
+}
+
+static void payload_count_flush_or_inc_on_close(struct lll_conn_iso_stream *cis_lll)
+{
+	if (cis_lll->tx.bn) {
+		uint64_t payload_count;
+		uint8_t u;
+
+		payload_count = cis_lll->tx.payload_count + cis_lll->tx.bn_curr - 1U;
+		u = cis_lll->nse - ((cis_lll->nse / cis_lll->tx.bn) *
+				    (cis_lll->tx.bn - 1U -
+				     (payload_count % cis_lll->tx.bn)));
+		if ((((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr +
+		      cis_lll->tx.ft + u) <=
+		     (cis_lll->event_count + cis_lll->nse + 2U)) ||
+		    (((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr) >
+		     (cis_lll->event_count + cis_lll->tx.bn))) {
+			uint8_t sn_update = cis_lll->tx.bn + 1U - cis_lll->tx.bn_curr;
+
+			/* sn and nesn are 1-bit, only Least Significant bit is needed */
+			cis_lll->sn += sn_update;
+			cis_lll->tx.payload_count += cis_lll->tx.bn;
+			cis_lll->tx.bn_curr = 1U;
+		}
+	}
+
+	if (cis_lll->rx.bn) {
+		uint64_t payload_count;
+		uint8_t u;
+
+		payload_count = cis_lll->rx.payload_count + cis_lll->rx.bn_curr - 1U;
+		u = cis_lll->nse - ((cis_lll->nse / cis_lll->rx.bn) *
+				    (cis_lll->rx.bn - 1U -
+				     (payload_count % cis_lll->rx.bn)));
+		if ((((cis_lll->rx.payload_count / cis_lll->rx.bn) + cis_lll->rx.bn_curr +
+		      cis_lll->rx.ft + u) <=
+		     (cis_lll->event_count + cis_lll->nse + 2U)) ||
+		    (((cis_lll->rx.payload_count / cis_lll->rx.bn) + cis_lll->rx.bn_curr) >
+		     (cis_lll->event_count + cis_lll->rx.bn))) {
+			uint8_t nesn_update = cis_lll->rx.bn + 1U - cis_lll->rx.bn_curr;
+
+			/* sn and nesn are 1-bit, only Least Significant bit is needed */
+			cis_lll->nesn += nesn_update;
+			cis_lll->rx.payload_count += cis_lll->rx.bn;
+			cis_lll->rx.bn_curr = 1U;
+		}
+	}
+}
+
+static void payload_count_lazy(struct lll_conn_iso_stream *cis_lll, uint16_t lazy)
+{
+	if (cis_lll->tx.bn) {
+		uint16_t tx_lazy;
+
+		tx_lazy = lazy;
+		while (tx_lazy--) {
+			uint64_t payload_count;
+			uint8_t u;
+
+			payload_count = cis_lll->tx.payload_count + cis_lll->tx.bn_curr - 1U;
+			u = cis_lll->nse - ((cis_lll->nse / cis_lll->tx.bn) *
+					    (cis_lll->tx.bn - 1U -
+					     (payload_count % cis_lll->tx.bn)));
+			if ((((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr +
+			      cis_lll->tx.ft + u) <
+			     (cis_lll->event_count + cis_lll->nse + 2U)) ||
+			    (((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr) >
+			     (cis_lll->event_count + cis_lll->tx.bn))) {
+				uint8_t sn_update = cis_lll->tx.bn + 1U - cis_lll->tx.bn_curr;
+
+				/* sn and nesn are 1-bit, only Least Significant bit is needed */
+				cis_lll->sn += sn_update;
+				cis_lll->tx.payload_count += cis_lll->tx.bn;
+				cis_lll->tx.bn_curr = 1U;
+			}
+		}
+	}
+
+	if (cis_lll->rx.bn) {
```

Comment:
Ditto

Reply:
Regarding refactoring, it will be addressed in #63152.